From: Tony Jones <tonyj@suse.de>
Subject: fix kabi due to perf_event.h uapi field change
Patch-mainline: Never, SLE15 kABI fix
References: FATE#325127

patches.suse/perf-x86-fix-data-source-decoding-for-skylake.patch changed
perf_mem_data_src union type.  Change is simply allocating current space
in padded reserve. Handle via __GENKSYMS__

Signed-off-by: Tony Jones <tonyj@suse.de>

---
 include/uapi/linux/perf_event.h |    8 ++++++++
 1 file changed, 8 insertions(+)

--- a/include/uapi/linux/perf_event.h
+++ b/include/uapi/linux/perf_event.h
@@ -931,20 +931,28 @@ union perf_mem_data_src {
 			mem_snoop:5,	/* snoop mode */
 			mem_lock:2,	/* lock instr */
 			mem_dtlb:7,	/* tlb access */
+#ifndef __GENKSYMS__
 			mem_lvl_num:4,	/* memory hierarchy level number */
 			mem_remote:1,   /* remote */
 			mem_snoopx:2,	/* snoop mode, ext */
 			mem_rsvd:24;
+#else
+			mem_rsvd:31;
+#endif
 	};
 };
 #elif defined(__BIG_ENDIAN_BITFIELD)
 union perf_mem_data_src {
 	__u64 val;
 	struct {
+#ifndef __GENKSYMS__
 		__u64	mem_rsvd:24,
 			mem_snoopx:2,	/* snoop mode, ext */
 			mem_remote:1,   /* remote */
 			mem_lvl_num:4,	/* memory hierarchy level number */
+#else
+		__u64	mem_rsvd:31,
+#endif
 			mem_dtlb:7,	/* tlb access */
 			mem_lock:2,	/* lock instr */
 			mem_snoop:5,	/* snoop mode */
